web/client/utils/FeatureGridUtils.js

```diff
@@ -162,19 +162,50 @@ module.exports = {
     applyAllChanges: (orig, changes = {}) => applyChanges(orig, changes[orig.id] || {}),
     applyChanges,
     gridUpdateToQueryUpdate: ({attribute, operator, value, type} = {}, oldFilterObj = {}) => {
+        const andGroup = {
+            id: 9999, // dummy number representing group id
+            logic: "AND",
+            index: 0
+        };
+
+        let mergeFilters = oldFilterObj.groupFields 
```

Comment:
Can you describe better the reason of these checks. I'm not sure but I think it could be simpler

Reply:
Those checks are used to know whether query panel filter is active or not(if active it will be merged with the Featuregrid filter). oldFilterObj object is used in the process of determining activeness

Reply:
Kindly adivice me what would be the best on this, I would like to reduce the checks, however some filter cases depend them.

Reply:
Please add some comments that clarify under what circumstances the merge should be applied.

Reply:
Merge should only be applied when there is at least one filter object coming from query panel filter inside the oldFilterObj